opsimulator/opsimulator.go

```diff
@@ -90,6 +100,37 @@ func (a *OpSimulator) Stopped() bool {
 	return a.stopped.Load()
 }
 
+func handler(proxy *httputil.ReverseProxy) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, "Failed to read request body", http.StatusInternalServerError)
+			return
+		}
+		defer r.Body.Close()
+
+		var req JSONRPCRequest
+		err = json.Unmarshal(body, &req)
+		if err != nil {
+			http.Error(w, "Failed to parse JSON-RPC request", http.StatusBadRequest)
+			return
+		}
+
+		if req.Method == "eth_sendRawTransaction" {
+			checkInteropInvariants()
```

Comment:
nit: can be in a separate PR, we should add the early return that write the result/response to ResponseWriter so it's easier to see how it would be used

Reply:
let's do that in a separate pr. I'll have to look more into what that would look like and I want to get this in before EOD, so it is not blocking.